### PR TITLE
Remove signing history from the chain validation state.

### PR DESCRIPTION
**Pull request by dnadales**
Consensus layer takes care of maintaining the history of signers. This was a
vestigial field from when we did these checks in the ledger layer.

### CODE REVIEW

**Review by nc6 — APPROVED** [review]

**Comment by mrBliss** [comment]
@nc6 What are (dis)advantages of reusing the `SigningHistory` instead of maintaining it ourselves (i.e. the `PBftChainState`)? What about performance?

File: cardano-ledger/test/Test/Cardano/Chain/Block/CBOR.hs [hunk 1]
```diff
@@ -250,20 +250,6 @@ ts_roundTripProofCBOR :: TSProperty
...
 
 
---------------------------------------------------------------------------------
--- SigningHistory
---------------------------------------------------------------------------------
-
--- TODO:
--- goldenSigningHistory :: Property
--- goldenSigningHistory =
---   goldenTestCBOR exampleSigningHistory "test/golden/cbor/block/SigningHistory"
```
Comment (dcoutts) [inline cardano-ledger/test/Test/Cardano/Chain/Block/CBOR.hs:260]: Presumably this also means this patch should also remove this golden test file.

**Review by dcoutts — APPROVED** [review]
As @mrBliss suggests, lets make a note here of why we decided that this bit makes more sense to implement in the consensus than in the ledger.

I think the answer is essentially that it's logically part of Permissive BFT, so it should be in consensus and not the ledger. But if someone more closely involved in the code would like to add a comment in this PR, that's probably good for posterity.

**Comment by nc6** [comment]
So, the proximal reason for removing this is that it was never used (because we didn't spot it) and hence only used in test code. The functionality is already implemented in consensus and so there's no need for it.

The deeper reason is that if the signing window is maintained in the ledger, we have to consider how to implement `anachronisticProtocolLedgerView` for it. And, indeed, under the current conception we cannot do this, since we assume that the parts of the state which have to be updated along with new headers are maintained in the consensus chain state.

File: cardano-ledger/test/Test/Cardano/Chain/Block/CBOR.hs (hunk 1 above)
Comment (dnadales) [reply cardano-ledger/test/Test/Cardano/Chain/Block/CBOR.hs:260]: Good one! We don't have that file. I guess it was deleted, or never created ... (and I like tautologies apparently :/ )

**Pull request merged**